### PR TITLE
Ensure consistency of kickstart script options that change operational mode of the script.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

This unifies handling of the (currently) six kickstart.sh options that change the operational mode of the script from the default of attempting to update or install the agent. In particular:

- They are now mutually exclusive, with only the last one specified being honored.
- A warning is now issued every time that one of these options overrides a previously specified option.
- The help text about these options has been updated to better describe them in detail, and to clearly offset them from the ‘normal’ kickstart options.

The affected options are:

- `--reinstall`
- `--reinstall-even-if-unsafe`
- `--reinstall-clean`
- `--uninstall`
- `--claim-only`
- `--repositories-only`
- `--prepare-offline-install-source`

##### Test Plan

The mutual exclusion and warnings for overrides can be tested by simply running the script with more than one of the above options specified on the command line. A separate warning should be issued for each overridden option, with the last option being the one that actually gets used.

The help text updates can be viewed either in the documentation, or by running the kickstart script from this PR with the `--help` option.

##### Additional Information

This PR also re-synchronizes the FreeBSD install documentation with the regular kickstart install documentation WRT the kickstart script options.

### CODE REVIEW

**Review by thiagoftsm — DISMISSED** [review]
All options worked as expected on `Debian 11`.

File: packaging/installer/kickstart.sh
Comment (MrZammler) [inline packaging/installer/kickstart.sh:198]: Hi Austin.

A couple of thoughts:

1) `--reinstall` & `--reinstall-clean`. Should be the same, besides the fact the latter will do a cleanup as well. But when there is not an existing install, they behave differently. Should they behave the same if no existing install is found?

2) `--claim-only`. I think it should fail if there is not an existing install ?

Also we print the cases for "No existing install" with a couple different ways, i.e. `If there is not an existing install` & `If there is no existing install`. We should use the same I think (The latter seems to me a bit "better") :-)

File: packaging/installer/kickstart.sh
Comment (MrZammler) [inline packaging/installer/kickstart.sh:928]: (Not of this PR but caught my eye): `proide` -> `provide`.

File: packaging/installer/kickstart.sh
Comment (MrZammler) [inline packaging/installer/kickstart.sh:1446]: Another little happy accident: `configuraion` -> `configuration`

**Comment by Ferroin** [comment]
Rebased to resolve merge conflict and pick up CI changes.

File: packaging/installer/kickstart.sh
Comment (thiagoftsm) [reply packaging/installer/kickstart.sh:965]: @Ferroin do you plan to fix messages with this PR or another?

Comment (Ferroin) [reply packaging/installer/kickstart.sh:965]: I’ve got plans to go over all the messages in detail in another PR at this point.

File: packaging/installer/kickstart.sh
Comment (thiagoftsm) [reply packaging/installer/kickstart.sh:198]: A final question @Ferroin , are you going to modify this PR according these suggestions? :thinking:

Comment (Ferroin) [reply packaging/installer/kickstart.sh:198]: @thiagoftsm I meant to respond regarding them, but just forgot.

---

@MrZammler 

Things currently behave as you mention because it simplifies the code to have them behave that way. I do intend to fix this eventually, but I would consider such changes to be orthogonal to this PR.

Good catch on the option descriptions though, I agree that those should be consistent, and that the shorter form  is preferable here.

**Comment by thiagoftsm** [comment]
Hello @Ferroin ,

I decided to do a different test on your PR, I ran the next two commands on an environment that already had `netdata` installed on `/`:

```sh
#  ./packaging/installer/kickstart.sh --reinstall
#  ./packaging/installer/kickstart.sh --uninstall
```

I observed that `kickstart` reinstalled netdata at `/opt`, but when I used the script to remove, it removed the files at `/` keeping the installation at `/opt` intact. I was expecting to remove the latest installation instead the previous. 

I am copying part of the logs for you:

```sh
  |.-.   .-.   .-.   .-.   .-.   .  netdata  .-.   .-.   .-.   .-.   .-.   .-
  |   '-'   '-'   '-'   '-'   '-'               '-'   '-'   '-'   '-'   '-'   
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

[/opt/netdata]# chmod 0644 /opt/netdata/etc/netdata/netdata.conf 
 OK  ''

 OK 

grep: warning: stray \ before -
Failed to enable unit: Unit file /usr/lib/systemd/system/netdata-updater.timer is masked.
 WARNING  Failed to enable auto updates. Netdata will still work, but you will need to update manually.

Successfully installed the Netdata Agent.

The following non-fatal warnings or errors were encountered:

  - We do not provide native packages for arch.
  - Could not install native binary packages, falling back to alternative installation method.
  - Failed to enable auto updates. Netdata will still work, but you will need to update manually.

Official documentation can be found online at https://learn.netdata.cloud/docs/.

Looking to monitor all of your infrastructure with Netdata? Check out Netdata Cloud at https://app.netdata.cloud.

Join our community and connect with us on:
  - GitHub: https://github.com/netdata/netdata/discussions
  - Discord: https://discord.gg/5ygS846fR6
  - Our community forums: https://community.netdata.cloud/
[root@archlinux netdata]# ./packaging/installer/kickstart.sh --uninstall

 --- Using /tmp/netdata-kickstart-nU5N1PoLx3 as a temporary directory. --- 
 --- Checking for existing installations of Netdata... --- 
 --- Found existing netdata-uninstaller. Running it.. --- 
[/tmp/netdata-kickstart-nU5N1PoLx3]# //usr/libexec/netdata/netdata-uninstaller.sh --yes 
Tue Nov 15 02:12:28 UTC 2022 : INFO: netdata-uninstaller.sh:  Stopping a possibly running netdata...
Do you want to delete this file '/etc/logrotate.d/netdata' ?  [y/n]



Tue Nov 15 02:13:19 UTC 2022 : ERROR: netdata-uninstaller.sh:  Failed to delete user netdata from group squid !
Tue Nov 15 02:13:19 UTC 2022 : INFO: netdata-uninstaller.sh:  Deleting netdata user from ceph group ...
[/tmp/netdata-kickstart-nU5N1PoLx3]# gpasswd -d netdata ceph 
gpasswd: group 'ceph' does not exist in /etc/group
 FAILED 

Tue Nov 15 02:13:19 UTC 2022 : ERROR: netdata-uninstaller.sh:  Failed to delete user netdata from group ceph !
Tue Nov 15 02:13:19 UTC 2022 : INFO: netdata-uninstaller.sh:  Deleting netdata user from nobody group ...
[/tmp/netdata-kickstart-nU5N1PoLx3]# gpasswd -d netdata nobody 
Removing user netdata from group nobody
 OK 

Do you want to delete 'netdata' system user ?  [y/n]
y
Tue Nov 15 02:13:21 UTC 2022 : INFO: netdata-uninstaller.sh:  Deleting netdata user account ...
[/tmp/netdata-kickstart-nU5N1PoLx3]# userdel -f netdata 
 OK 

Do you want to delete 'netdata' system group ?  [y/n]
y
Tue Nov 15 02:13:22 UTC 2022 : INFO: netdata-uninstaller.sh:  Removing netdata user group ...
Tue Nov 15 02:13:22 UTC 2022 : INFO: netdata-uninstaller.sh:  Group netdata already removed in a previous step.

Tue Nov 15 02:13:22 UTC 2022 : INFO: netdata-uninstaller.sh:  Netdata files were successfully removed from your system
 OK 

Finished uninstalling the Netdata Agent.[root@archlinux netdata]# ls -l /opt/netdata/
total 28
drwxrwxr-x 1 973 973 134 Nov 14 00:23 bin
drwxrwxr-x 1 973 973  20 Nov 15 02:11 etc
drwxr-xr-x 1 973 973   8 Nov 14 00:20 include
drwxr-xr-x 1 973 973  56 Nov 14 00:20 lib
lrwxrwxrwx 1 973 973  11 Nov 15 02:11 netdata-configs -> etc/netdata
lrwxrwxrwx 1 973 973  15 Nov 15 02:11 netdata-dbs -> var/lib/netdata
lrwxrwxrwx 1 973 973  15 Nov 15 02:11 netdata-logs -> var/log/netdata
lrwxrwxrwx 1 973 973  17 Nov 15 02:11 netdata-metrics -> var/cache/netdata
lrwxrwxrwx 1 973 973  19 Nov 15 02:11 netdata-plugins -> usr/libexec/netdata
lrwxrwxrwx 1 973 973  21 Nov 15 02:11 netdata-web-files -> usr/share/netdata/web
lrwxrwxrwx 1 973 973   3 Nov 15 02:11 sbin -> bin
drwxrwxr-x 1 973 973  40 Nov 14 00:23 share
drwxrwxr-x 1 973 973 276 Nov 14 00:23 system
drwxrwxr-x 1 973 973  54 Nov 15 02:11 usr
drwxrwxr-x 1 973 973  28 Nov 14 00:23 var
[root@archlinux netdata]#
```

**Comment by Ferroin** [comment]
> Hello @Ferroin ,
> 
> I decided to do a different test on your PR, I ran the next two commands on an environment that already had `netdata` installed on `/`:
> 
> ```shell
> #  ./packaging/installer/kickstart.sh --reinstall
> #  ./packaging/installer/kickstart.sh --uninstall
> ```
> 
> I observed that `kickstart` reinstalled netdata at `/opt`, but when I used the script to remove, it removed the files at `/` keeping the installation at `/opt` intact. I was expecting to remove the latest installation instead the previous.

The first part (reinstalling in `/opt` when an existing install is already present in `/`) is a bug, though possibly not one specific to this PR (I’ll have to dig a bit further.

The second part is expected behavior (if left to autodetect an existing install, the uninstall code uninstalls the first one it finds, and `/` gets searched before any other locations).

**Review by DShreve2 — APPROVED** [review]

**Comment by DShreve2** [comment]
Good from the docs perspective. Just an FYI, we'll be soft-freezing docs on Sunday so we don't lose anything in setting up the new structure next week. If we want to push this before then, no issue there, but if we need to merge next week, just let me know directly (I'll keep an eye on it).

**Comment by Ferroin** [comment]
Rebased to resolve merge conflicts and pick up CI changes.

**Comment by Ferroin** [comment]
Rebased to resolve merge conflicts and pick up CI changes.

**Review by thiagoftsm — DISMISSED** [review]
This PR is working as expected on:

- Slackware Current
- FreeBSD 13.1
- Debian 11
- CentOS 7

LGTM!

**Comment by thiagoftsm** [comment]
> The first part (reinstalling in `/opt` when an existing install is already present in `/`) is a bug, though possibly not one specific to this PR (I’ll have to dig a bit further.

@Ferroin I did not observe the issue I reported.

**Review by thiagoftsm — DISMISSED** [review]

**Review by cakrit — APPROVED** [review]

**Comment by Ferroin** [comment]
Rebased to resolve merge conflict.

**Comment by Ferroin** [comment]
Given that:

- The last rebase made no functional changes.
- Multiple approvals were made prior to the rebase.
- All relevant CI was passing prior to the rebase.
- I’ve verified the merge that was done as part of the rebase as being correct multiple times locally.
- This has been open for six months, and is starting to become a blocker for _other_ improvements to the kickstart script.

I’m just going to merge it without waiting for CI or approvals.

**Pull request merged**
